realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,776 @@
+     * @throws IllegalArgumentException if the requested column is not a double.
+     */
+    @Override
+    public double getDouble(int columnIndex) {
+        checkClosed();
+        checkPosition();
+        return table.getDouble(columnIndex, rowIndex);
+    }
+
+    /**
+     * Returns the SQLite Cursor type for a column index. Realm has more types than SQLite and some of them cannot be
+     * mapped to any meaningful SQLite type. These will return -1 and cannot be accessed using a cursor.
+     *
+     * The following types are currently not supported:
+     * <ul>
+     *   <li>{@link ColumnType#TABLE}</li>
+     *   <li>{@link ColumnType#MIXED}</li>
+     *   <li>{@link ColumnType#LINK}</li>
+     *   <li>{@link ColumnType#LINK_LIST}</li>
+     * </ul>
+     *
+     * <p>
+     * Some realm types are mapped to conform to the Cursor interface. These mappings are described below:
+     * <ol>
+     *   <li>boolean : Will return Cursor.FIELD_TYPE_INTEGER. {0 = false, 1 = true}, use getShort()/getInt()/getLong()</li>
+     *   <li>double : Will return Cursor.FIELD_TYPE_FLOAT, but must be fetched using use getDouble()</li>
+     *   <li>Date : Will return Cursor.FIELD_TYPE_INTEGER. Time in milliseconds since epoch, use getLong()</li>
+     *</ol>
+     *
+     * @param columnIndex Get the data type for this column index or -1 if the column isn't accessible.
+     * @return One of the {@code FIELD_TYPE_*}'s described in {@link Cursor}.
+     */
+    @Override
+    public int getType(int columnIndex) {
+        checkClosed();
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch (realmType) {
+            case BOOLEAN: return FIELD_TYPE_INTEGER;
+            case INTEGER: return FIELD_TYPE_INTEGER;
+            case FLOAT: return FIELD_TYPE_FLOAT;
+            case DOUBLE: return FIELD_TYPE_FLOAT;
+            case STRING: return FIELD_TYPE_STRING;
+            case BINARY: return FIELD_TYPE_BLOB;
+            case DATE: return FIELD_TYPE_INTEGER;
+            case TABLE:
+            case MIXED:
+            case LINK:
+            case LINK_LIST:
+            default:
+                return -1;
+        }
+    }
+
+    /**
+     * Realm currently doesn't support null. Calling this method will always throw an exception.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public boolean isNull(int columnIndex) {
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch(realmType) {
+            case LINK: return table.isNullLink(columnIndex, rowIndex);
+            case LINK_LIST:
+            case BOOLEAN:
+            case INTEGER:
+            case FLOAT:
+            case DOUBLE:
+            case STRING:
+            case BINARY:
+            case DATE:
+            case TABLE:
+            case MIXED:
+            default:
+            throw new UnsupportedOperationException("isNull not yet supported by Realm for this type: " + realmType);
```

Comment:
It is. I am not a big fan of using @Since/@Version before 1.0 as the API is still pr. definition unstable at that point, but those annotations are definitely worth keeping in mind.